safe_landing_planner/include/safe_landing_planner/grid.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+namespace avoidance {
+
+class Grid {
+public:
+  Grid(const float grid_size, const float cell_size) :
+    grid_size_(grid_size),
+    cell_size_(cell_size),
+    mean_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size))),
+    variance_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size))),
+    counter_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size))),
+    land_(static_cast<int>(std::ceil(grid_size/cell_size)), static_cast<int>(std::ceil(grid_size/cell_size)))
+    {
+      reset();
+    }
+  ~Grid() = default;
+
+  void reset() {
+    mean_.fill(0.f);
+    variance_.fill(0.f);
+    counter_.fill(0);
+    land_.fill(0);
+  }
+
+  void resize() {
```

Comment:
This should take the new size as an argument, rather than letting the sizes be publicly available. Otherwise the size variables might be out of sync with the grids.